### ISSUE TITLE
DevTools not working with Version 0.0.16

### ISSUE THREAD

**Issue by mbartels** [MarcGiffing/wicket-spring-boot#77]
Hi Marc,

I am using the current version 0.0.16 with Spring DevTools and it is not working anymore. I think the reason is the introduction of the class: "SpringDevToolsProperties"

Without the class in version 0.0.15 my app starts as expected. With the new version I get the following exception:

`Caused by: org.springframework.beans.factory.BeanCreationException: Could not autowire field: private com.giffing.wicket.spring.boot.starter.configuration.extensions.external.development.springboot.devtools.SpringDevToolsProperties com.giffing.wicket.spring.boot.starter.configuration.extensions.external.development.springboot.devtools.SpringDevtoolsSerializerConfig.props; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [com.giffing.wicket.spring.boot.starter.configuration.extensions.external.development.springboot.devtools.SpringDevToolsProperties] found for dependency: expected at least 1 bean which qualifies as autowire candidate for this dependency. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:573)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:88)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessPropertyValues(AutowiredAnnotationBeanPostProcessor.java:331)
	... 22 common frames omitted
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [com.giffing.wicket.spring.boot.starter.configuration.extensions.external.development.springboot.devtools.SpringDevToolsProperties] found for dependency: expected at least 1 bean which qualifies as autowire candidate for this dependency. Dependency annotations: {@org.springframework.beans.factory.annotation.Autowired(required=true)}
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.raiseNoSuchBeanDefinitionException(DefaultListableBeanFactory.java:1373)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1119)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1014)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:545)
	... 24 common frames omitted`

Somehow Spring is not able to find the the property class.

Best regards,
Mat

**Comment by MarcGiffing** [comment]
Are you sure that this is the first thrown exception? I couldn't see a reason why this bean shouldn't be created. I will provide a simple example project to reproduce the problem later this day.

**Comment by mbartels** [comment]
It looks strange to me as well, because your code change for the DevTools-AutoConfig looked fine for me
 
But as I said switching back to 0.0.15 solves the problem without any other modification and these are the only Exceptions I can see.

**Comment by MarcGiffing** [comment]
Would you please checkout the following project to check if get the same exception. Thanks!
https://github.com/MarcGiffing/wicket-spring-boot-examples/tree/master/devtools

**Comment by mbartels** [comment]
Thanks for the example. The example works without exception. 
I investigated a little further and realized some version differences between your example and my project (wicket 7.3.0 vs. 7.4.0, spring boot 1.3.3 vs. 1.4.0). After playing around a little bit with the versions in my project I found out that an upgrade to spring boot 1.4.0 solved the problem!

That does not explain why it didn't work before, but solves my immediate problem. 
From my point of view we could close the bug.

Thanks for your fast response,
Mat
